src/circuit/gadget/utilities.rs

```diff
@@ -131,7 +131,7 @@ pub fn bitrange_subset<F: FieldExt + PrimeFieldBits>(field_elem: F, bitrange: Ra
 /// i.e. 0 ≤ word < range.
 pub fn range_check<F: FieldExt>(word: Expression<F>, range: usize) -> Expression<F> {
     (1..range).fold(word.clone(), |acc, i| {
-        acc * (word.clone() - Expression::Constant(F::from_u64(i as u64)))
+        acc * (Expression::Constant(F::from_u64(i as u64)) - word.clone())
```

Comment:
I noticed that this commit didn't include a book change, and that's because this implementation was previously conflicting with the [book](https://zcash.github.io/orchard/design/circuit/gadgets/decomposition.html#short-range-decomposition) (that definition was originally added in https://github.com/zcash/orchard/commit/2dd23f47b80da03c202f7cad2d04e7494ca7e433). It now matches the book, but we should verify that all uses of `range_check` (for which the circuit changes in this commit) match all uses of `range_check` in the book.